testkit/src/lib.rs

```diff
@@ -634,29 +624,52 @@ impl TestKit {
     ///
     /// # Panics
     ///
-    /// - Panics in the case any of transaction hashes are not in the mempool.
+    /// - Panics in the case any of transaction hashes are not in the pool.
     pub fn create_block_with_tx_hashes(&mut self, tx_hashes: &[crypto::Hash]) {
         self.poll_events();
 
         {
-            let txs = self.mempool();
+            let snapshot = self.blockchain.snapshot();
+            let schema = CoreSchema::new(&snapshot);
             for hash in tx_hashes {
-                assert!(txs.contains_key(hash));
+                assert!(schema.transactions_pool().contains(hash));
             }
         }
 
         self.do_create_block(tx_hashes);
     }
 
-    /// Creates block with all transactions in the mempool.
+    /// Creates block with all transactions in the pool.
     pub fn create_block(&mut self) {
         self.poll_events();
 
-        let tx_hashes: Vec<_> = self.mempool().keys().cloned().collect();
-
+        let snapshot = self.blockchain.snapshot();
+        let schema = CoreSchema::new(&snapshot);
+        let txs = schema.transactions_pool();
+        let tx_hashes: Vec<_> = txs.iter().collect();
+        //FIXME: every block should contain two merges
+        {
+            let blockchain = self.blockchain_mut();
+            let fork = blockchain.fork();
+            blockchain.merge(fork.into_patch()).unwrap();
+        }
         self.do_create_block(&tx_hashes);
     }
 
+    /// Adds transaction into persistent pool.
+    pub fn add_tx(&mut self, transaction: RawMessage) {
```

Comment:
I don't think that there is a need for such a method; it seems to duplicate `testkit.api().send(transaction)`. (Also, it can probably break some invariants, such as putting non-verifiable or confirmed transactions into the pool; although, admittedly, I haven't analyzed this precisely.)